controllers/subject.js

```diff
@@ -1,7 +1,12 @@
 const subjectService = require('~/services/subject')
+const getRegex = require('~/utils/getRegex')
 
 const getSubjects = async (req, res) => {
-  const subjects = await subjectService.getSubjects()
+  const { skip, limit, name } = req.query
+
+  const searchFilter = { name: getRegex(name) }
+  
+  const subjects = await subjectService.getSubjects({ skip:parseInt(skip), limit:parseInt(limit), searchFilter })
```

Comment:
i guess, better to get props not as object in service, and here, you can pass them without skip: ...., limit: ....

Reply:
good, but as for me passing like an object can avoid errors cause by not right sequence of arguments